### ISSUE TITLE
Shapes on Android don't support Horizontal/VerticalOptions

### ISSUE THREAD

**Issue by albi005** [dotnet/maui#9433]
### Description

The Shape uses the all of the available space, instead of being sized based on its content.

This only happens on Android. It seems like the problem comes from https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/Shapes/Shape.cs#L259,
where the return value of `base.MeasureOverride(widthConstraint, heightConstraint)` (the available space, `widthConstraint` and `heightConstraint`) is returned instead of the size of the Path that is supposed to be calculated later on.

### Steps to Reproduce

1. Create a new .NET MAUI app
2. Replace MainPage's Content with
```xaml
    <Path
        BackgroundColor="Blue"
        Data="M8.464 10.606v16.065h20V10.606l-10-7.935z"
        HorizontalOptions="Center"
        Stroke="LightBlue"
        StrokeThickness="4"
        VerticalOptions="Center" />
```
3. The Path is supposed to be centered but its not

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

Setting HeightRequest and WidthRequest to 0 can be used as a workaround, but this breaks the Windows version.

**Comment by XceedBoucherS** [comment]
I have the same problem on Android. 
I would like to not have to set a WidthRequest/HeightRequest to 0.

Thank you.

**Comment by hartez** [comment]
Fixed by #6206.
